### PR TITLE
Dark mode fixed

### LINKED ISSUE #110: Fix secure cookie not able to update from client
Change to updating cookie server side instead of client since cookie is secure for purposes of keeping it alive on phones

### PR DESCRIPTION
**Pull request by carlhampuswall**
Should now work cookies are saved trough server and secure, fixed small issue with getInitalProps in _app returning unnecessary amount of undefined

Closes #110

### CODE REVIEW

File: src/hooks/use-color-scheme.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import { ColorScheme } from '@mantine/core';
```
Comment (carlhampuswall) [inline src/hooks/use-color-scheme.ts:1]: Added server call to setColoScheme to handle setting secure cookie

File: src/hooks/use-color-scheme.ts [hunk 2]
```diff
@@ -9,12 +10,8 @@ const useColorScheme = (initialColorScheme: ColorScheme) => {
     const newColorScheme =
       value || (colorScheme === 'light' ? 'dark' : 'light');
     setColorScheme(newColorScheme);
-    setCookie('mantine-color-scheme', newColorScheme, {
-      secure: true,
-      sameSite: 'lax',
-      httpOnly: true,
-    });
   };
+  trpc.corps.setColorScheme.useQuery(colorScheme);
```
Comment (carlhampuswall) [inline src/hooks/use-color-scheme.ts:14]: Added server call to setColoScheme to handle setting secure cookie

File: src/pages/_app.tsx [hunk 3]
```diff
@@ -64,6 +65,10 @@ MyApp.getInitialProps = async (context: AppContext) => {
   const { ctx } = context;
   const appProps = await App.getInitialProps(context);
   const cookie = ctx.req?.headers.cookie;
+  if (ctx.req?.headers.cookie === undefined) {
```
Comment (carlhampuswall) [inline src/pages/_app.tsx:68]: Removes unnecessary returning of undefined

File: src/server/trpc/router/corps.ts [hunk 4]
```diff
@@ -266,76 +278,87 @@ export const corpsRouter = router({
...
+  setColorScheme: protectedProcedure
+    .input(z.string())
+    .query(async ({ ctx, input }) => {
+      const { req, res } = { req: ctx.req, res: ctx.res };
+      setCookie('mantine-color-scheme', input, {
+        req,
+        res,
+        expires: new Date(Date.now() + 1000 * 60 * 60 * 24 * 365),
+      });
+      return { success: true };
```
Comment (carlhampuswall) [inline src/server/trpc/router/corps.ts:362]: Added function that sets cookie server side

File: src/server/trpc/router/corps.ts [hunk 5]
```diff
@@ -266,76 +278,87 @@ export const corpsRouter = router({
...
+  }),
+
+  setColorScheme: protectedProcedure
+    .input(z.string())
+    .query(async ({ ctx, input }) => {
+      const { req, res } = { req: ctx.req, res: ctx.res };
+      setCookie('mantine-color-scheme', input, {
+        req,
+        res,
+        expires: new Date(Date.now() + 1000 * 60 * 60 * 24 * 365),
```
Comment (CodeyBoi) [inline src/server/trpc/router/corps.ts:360]: Please make the cookie never expire.

File: src/pages/_app.tsx [hunk 6]
```diff
@@ -15,6 +15,7 @@ import Head from 'next/head';
 import App from 'next/app';
 import cookieParser from '../utils/cookie-parser';
 import { ColorScheme } from '@mantine/core';
+import { useEffect } from 'react';
```
Comment (CodeyBoi) [inline src/pages/_app.tsx:18]: Unused import

**Review by CodeyBoi — CHANGES_REQUESTED** [review]

**Review by CodeyBoi — APPROVED** [review]

**Pull request merged**